ClinicalTrial_SE_curation.Rmd

```diff
@@ -0,0 +1,320 @@
+                       survival_type = NA, survival_time = NA, survival_time_unit= NA, event_occured = NA)
+
+colData(GSE50948_SE)[["cancer_type"]] <- "Breast cancer"
+
+```
+
+```{r}
+#==== 7 ====
+GSE33072_SE <- run_all(raw_tar = "GSE33072_RAW.tar" , 
+                       geo_path = "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE33072&format=file" , 
+                       brain_path = "http://mbni.org/customcdf/24.0.0/ensg.download/hugene10sthsensgcdf_24.0.0.tar.gz",
+                       delim = "_.*" , GPL = "GPL6244", delim2=".CEL.*",cancer_type=NA, treatment = "treatment:ch1", response = NA, 
+                       survival_type = NA, survival_time = NA, survival_time_unit= NA, event_occured = NA)
+
+colData(GSE33072_SE)[["cancer_type"]] <- "Lung cancer"
+colData(GSE33072_SE) [["survival_type"]] <- "progression_free_survival"
+colData(GSE33072_SE)$survival_time <- ifelse(is.na(colData(GSE33072_SE)$progression.free.survival.time..months..ch1), colData(GSE33072_SE)$pfsm..month..ch1, colData(GSE33072_SE)$progression.free.survival.time..months..ch1)
+colData(GSE33072_SE)$event_occured <- ifelse(is.na(colData(GSE33072_SE)$progression.free.survival.status.ch1), colData(GSE33072_SE)$pfsc..1.progressed..0.not.progressed..ch1, colData(GSE33072_SE)$progression.free.survival.status.ch1)
+colData(GSE33072_SE)[["survival_time_unit"]] <- "month"
+
+```
+
+```{r}
+#==== 8 ====
+GSE25066_SE <- run_all(raw_tar = "GSE25066_RAW.tar" , 
+                       geo_path = "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE25066&format=file" , 
+                       brain_path = "http://mbni.org/customcdf/24.0.0/ensg.download/hgu133ahsensgcdf_24.0.0.tar.gz",
+                       delim = "_.*" , GPL = "GPL96", cancer_type=NA,treatment = NA, response = "pathologic_response_pcr_rd:ch1" , 
+                       survival_type = NA, survival_time = "drfs_even_time_years:ch1",  
+                       survival_time_unit= NA, event_occured = "drfs_1_event_0_censored:ch1")
+
+colData(GSE25066_SE)[["cancer_type"]] <- "Breast cancer"
+colData(GSE25066_SE)[["treatment"]] <- "taxane-anthracycline"
+colData(GSE25066_SE)[["survival_type"]] <- "Distant recurrence-free survival (DRFS)"
+colData(GSE25066_SE)[["survival_time_unit"]] <- "year"
+
+```
+
+```{r}
+#==== 9 ====
+GSE15622_SE <- run_all(raw_tar = "GSE15622_RAW.tar" , 
+                       geo_path = "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE15622&format=file" , 
+                       brain_path = "http://mbni.org/customcdf/24.0.0/ensg.download/hgu133a2hsensgcdf_24.0.0.tar.gz",
+                       delim = ".CEL" , GPL = "GPL8414", cancer_type=NA,treatment = "treatment:ch1", response = "response:ch1" , 
+                       survival_type = NA, survival_time = NA,  
+                       survival_time_unit= NA, event_occured = NA)
+
+colData(GSE15622_SE)[["cancer_type"]] <- "Ovarian Cancer"
+```
+
+
+```{r}
+#===== Save objects =======
```

Comment:
Suggestion:

This should probably be a for loop:
```
gse_datasets <- c("GSE41998_SE", "GSE14671_SE", ...)
for (dataset in gse_datasets) qsave(get(dataset), file=paste0(dataset, ".qs")
```

Also, `qsave` really isn't any better than `saveRDS` unless you set the `nthread` parameter like:
```
nthread <- 2
qsave(..., nthread=nthread)
```
This is also true for `qread`.

You can set the nthread as a variable at the top of your script to make it easy to change the threading everywhere.

Reply:
@NiktaFeizi

Reply:
Done